pkg/shell/active-pages-modal.jsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+
+import React from "react";
+import { ListingTable } from "cockpit-components-table.jsx";
+import { Button, Label, Split, SplitItem, Modal } from '@patternfly/react-core';
+
+const _ = cockpit.gettext;
+
+export class ActivePagesDialog extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const frames = [];
+        for (const address in props.frames.iframes) {
+            for (const component in props.frames.iframes[address]) {
+                const iframe = props.frames.iframes[address][component];
+                frames.push({
+                    frame: iframe,
+                    component: component,
+                    address: address,
+                    name: iframe.getAttribute("name"),
+                    visible: iframe.getAttribute("data-active") === 'true',
+                    selected: iframe.getAttribute("data-active") === 'true',
```

Comment:
That looks weird -- there's only ever one active frame, isn't it? Why are all the others initialized as non-visible? I may misunderstand what data-active means of course, and the old code initialized `visible` like it does here. Just the `selected` is new.

Reply:
`s/visible/active` I guess. I just moved it, but I can rename it. Selected means if you checked that frame in the modal to be closed. By default active page is selected.